gammapy/datasets/map.py

```diff
@@ -1841,29 +1842,35 @@ def cutout(self, position, width, mode="trim", name=None):
             Cutout map dataset.
         """
         name = make_name(name)
-        kwargs = {"gti": self.gti, "name": name, "meta_table": self.meta_table}
-        cutout_kwargs = {"position": position, "width": width, "mode": mode}
```

Comment:
You can keep `cutout_kwargs` and directly set `kwargs["meta"].optional = cutout_kwargs` to avoid the later changes

Reply:
Yes but is there some reason to have the `cutout_kwargs`? They cannot be accessed from outside

Reply:
Using `cutout_kwargs` is syntactically shorter, when the same keywords are passed to multiple methods. Passing arguments by name is usually preferred to passing by position, because its is more explicit. However using `.cutout(position=position, width=width, mode=mode)` all over again is cumbersome. Hence the `kwargs` provides the cleanest solution.